multinode_integration_tests/src/main.rs

```diff
@@ -342,6 +349,104 @@ impl MockNode {
     }
 }
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+struct ProbeTarget {
+    ip_address: IpAddr,
+    port_specs: Vec<PortSpec>,
+}
+
+impl FromStr for ProbeTarget {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut main_pieces = s.split(':').collect_vec();
+        if main_pieces.len() != 2 {
+            return Err("Syntax: <IP address>:<U|T><port>/<U|T><port>/...".to_string());
+        }
+        let ip_address_string = main_pieces.remove(0);
+        let port_specs_string = main_pieces.remove(0);
```

Comment:
In theory, you did not have to make the vector mutable and remove the items out.  Because all you need are referenced pieces of &str you might have been able to use just the slice indexing like `main_pieces[0]`. But let's perhaps not make this test code as dry as spots in Sahara.

Reply:
I kind of like consuming the pieces; it seems more Rust-like than letting them lie around and referencing them.

Reply:
That's legit. :-)